standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/AcidMetricService.java

```diff
@@ -85,36 +85,113 @@ public void run() {
 
   private void collectMetrics() throws MetaException {
     ShowCompactResponse currentCompactions = txnHandler.showCompact(new ShowCompactRequest());
-    updateMetricsFromShowCompact(currentCompactions);
+    updateMetricsFromShowCompact(currentCompactions, conf);
     updateDBMetrics();
   }
 
   private void updateDBMetrics() throws MetaException {
     MetricsInfo metrics = txnHandler.getMetricsInfo();
     Metrics.getOrCreateGauge(NUM_TXN_TO_WRITEID).set(metrics.getTxnToWriteIdCount());
+    if (metrics.getTxnToWriteIdCount() >=
```

Comment:
It would be more readable to separate out all the logging into a new method (can just pass the MetricsInfo object)